src/Elastic.Transport/Components/Serialization/TransportSerializerExtensions.cs

```diff
@@ -72,4 +77,304 @@ public static string SerializeToString<T>(
 		serializer.Serialize(data, ms, formatting);
 		return ms.Utf8String();
 	}
+
+	#region STJ Extensions
```

Comment:
STJ specific extension methods that implement efficient serialization to:

- `Utf8JsonWriter`

and deserialization from:

- `Utf8JsonReader`
- `JsonNode`
- `JsonElement`